pkg/storage/upload.go

```diff
@@ -47,7 +74,67 @@ func getResumableUploadUrl(url string) (string, error) {
 		return "", err
 	}
 
-	defer res.Body.Close()
-
 	return res.Header.Get("Location"), nil
 }
+
+var backoffSchedule = []time.Duration{
+	1 * time.Second,
+	2 * time.Second,
+	5 * time.Second,
+}
+
+func uploadChunkWithRetryPolicy(
+	client *http.Client,
+	uploadUrl string,
+	chunk []byte,
+	offset int64,
+	fileSize int64,
+) error {
+	var err error
+
+	for _, backoff := range backoffSchedule {
+		err = uploadChunk(client, uploadUrl, chunk, offset, fileSize)
+		if err == nil {
+			break
+		}
+
+		time.Sleep(backoff)
+	}
+
+	return err
+}
+
+func uploadChunk(
+	client *http.Client,
+	uploadUrl string,
+	chunk []byte,
+	offset int64,
+	fileSize int64,
+) error {
+	req, err := http.NewRequest("PUT", uploadUrl, bytes.NewReader(chunk))
+	if err != nil {
+		return err
+	}
+
+	chunkSize := int64(len(chunk))
+	byteRange := fmt.Sprintf("bytes %d-%d/%d", offset, offset+chunkSize-1, fileSize)
+	lastChunk := offset+defaultChunkSize >= fileSize
+
+	req.Header.Set("Content-Length", fmt.Sprint(chunkSize))
+	req.Header.Set("Content-Range", byteRange)
+
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
```

Comment:
Todo
Can we pass thru an instance of http client so we can configure it later

Reply:
done - completed in another PR